keras/utils/np_utils.py

```diff
@@ -48,3 +49,62 @@ def normalize(x, axis=-1, order=2):
     l2 = np.atleast_1d(np.linalg.norm(x, order, axis))
     l2[l2 == 0] = 1
     return x / np.expand_dims(l2, axis)
+
+def mxnet_image_channels_conversion(np_data):
+    """
+    Transform the image tensor data from channels_last to channels_first
+    if using MXNet backend
+
+    # Arguments
+        np_data: image data tensor
+
+    # Returns
+        A image data tensor with channels_first format
+    """
+    data_format = K.image_data_format()
+    if data_format not in {'channels_first', 'channels_last'}:
+        raise ValueError('MXNet Backend: Unknown data_format ' +
+                         str(data_format))
+    if K.backend() == 'mxnet':
+        if data_format == 'channels_first':
+            shape = np_data.shape
+            if len(shape) == 5:
+                if shape[1] < shape[4]:
+                    warnings.warn('Image data tensor format is already '
+                                  '`channels_first`', stacklevel=2)
+                    return np_data
+                np_data = np.transpose(np_data, (0, 4, 1, 2, 3))
+            elif len(shape) == 4:
+                if shape[1] < shape[3]:
```

Comment:
what if I have an image of 2*2 ?

Reply:
The method was written in such a way that if user calls `to_channels_first()` two or more times accidently on the same data tensor, then the method should return the ouput in always `channels_first` format. If I remove that check, then there is a channel conversion mismatch if user calls `to_channels_first()` multiple times.

```python
For instance:

Before check:
x_input = A Numpy tensor of shape (10, 3, 32, 32)
x_input = to_channels_first(x_input) -> output shape (10, 32, 3, 32)

After check:
x_input = A Numpy tensor of shape (10, 3, 32, 32)
x_input = to_channels_first(x_input) -> output shape (10, 3, 32, 32)
```

So, by keeping that shape check, we won't face such issues. However, as u mentioned, it will fail when image dimension is `(2, 2)` or `(1, 1)` or values smaller than channels in general. This would be possible if the data is synthetic and also, there is no other way to check based on tensors dimensions and impossible to cover all the edge cases. So, I have removed the check and make it more generic and assume user aware of it's data dimensions.

Thank-You.